lib/services/local/mfa_test.go

```diff
@@ -0,0 +1,231 @@
+// Teleport
+// Copyright (C) 2025 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package local_test
+
+import (
+	"cmp"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+
+	mfav1 "github.com/gravitational/teleport/api/gen/proto/go/teleport/mfa/v1"
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/backend/memory"
+	local "github.com/gravitational/teleport/lib/services/local"
+)
+
+func TestMFAService_CRUD(t *testing.T) {
+	t.Parallel()
+
+	backend, err := memory.New(memory.Config{})
+	require.NoError(t, err)
+
+	svc, err := local.NewMFAService(backend)
+	require.NoError(t, err)
+
+	username, chal := "alice", newValidatedMFAChallenge()
+
+	// Create
+	created, err := svc.CreateValidatedMFAChallenge(t.Context(), username, chal)
+	require.NoError(t, err)
+	require.Equal(t, chal.Kind, created.Kind)
+	require.Equal(t, chal.Version, created.Version)
+	require.Equal(t, chal.Metadata.Name, created.Metadata.Name)
+	require.WithinDuration(t, time.Now().Add(5*time.Minute), *created.Metadata.Expires, time.Second)
+
+	// Get
+	got, err := svc.GetValidatedMFAChallenge(t.Context(), username, chal.Metadata.Name)
+	require.NoError(t, err)
+	require.Equal(t, created.Metadata.Name, got.Metadata.Name)
+	require.Equal(t, created.Spec.SourceCluster, got.Spec.SourceCluster)
+	require.Equal(t, created.Spec.TargetCluster, got.Spec.TargetCluster)
+	require.WithinDuration(t, *created.Metadata.Expires, *got.Metadata.Expires, time.Second)
+
+	// Get non-existent
+	_, err = svc.GetValidatedMFAChallenge(t.Context(), username, "does-not-exist")
+	require.ErrorContains(t, err, "doesn't exist")
```

Comment:
I'd also move this to the top.

```suggestion
	t.Run("get non-existent", func(t *testing.T) {
		t.Parallel()

		_, err = svc.GetValidatedMFAChallenge(t.Context(), username, "does-not-exist")
		assert.ErrorContains(t, err, "doesn't exist")
	})
```